[Lens] Formula time shift

flash1293 opened the pull request:
Fixes https://github.com/elastic/kibana/issues/35043

Adds time shift support to formula.

You can use time shift on all Elasticsearch functions (`average`, `sum` etc.) - the parameter is called `shift` and takes the same input as the "Time shift" advanced option for quick functions. Inputs than won't parse (like `3ds` instead of `3d`) will cause a formula error and won't render. Time shifts which are not aligned with the date histogram interval will cause a formula warning but will be treated as value and render as usual.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130042/)
* Commit: e350d196d0b32add07b834e8c6378174a19ba4c9
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Telemetry Schema](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130042/execution/node/415/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130042/","number":"130042","commit":"e350d196d0b32add07b834e8c6378174a19ba4c9"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130052/)
* Commit: 4b192bbaa731f93683194bdb5fab5e6d8a8f4b74
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130052/execution/node/428/log)

### History
* :broken_heart: [Build #130042](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130042/) failed e350d196d0b32add07b834e8c6378174a19ba4c9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130042/","number":"130042","commit":"e350d196d0b32add07b834e8c6378174a19ba4c9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130052/","number":"130052","commit":"4b192bbaa731f93683194bdb5fab5e6d8a8f4b74"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130087/)
* Commit: 75b8da7982416bedef0078116a0e6a04c1b65d8d
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130087/execution/node/421/log)

### History
* :broken_heart: [Build #130052](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130052/) failed 4b192bbaa731f93683194bdb5fab5e6d8a8f4b74
* :broken_heart: [Build #130042](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130042/) failed e350d196d0b32add07b834e8c6378174a19ba4c9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130042/","number":"130042","commit":"e350d196d0b32add07b834e8c6378174a19ba4c9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130052/","number":"130052","commit":"4b192bbaa731f93683194bdb5fab5e6d8a8f4b74"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130087/","number":"130087","commit":"75b8da7982416bedef0078116a0e6a04c1b65d8d"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130093/)
* Commit: d1f9231909e8e191114dce565cabe165373cd75b
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101718/d1f9231909e8e191114dce565cabe165373cd75b)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130093/execution/node/459/log)

### History
* :broken_heart: [Build #130087](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130087/) failed 75b8da7982416bedef0078116a0e6a04c1b65d8d
* :broken_heart: [Build #130052](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130052/) failed 4b192bbaa731f93683194bdb5fab5e6d8a8f4b74
* :broken_heart: [Build #130042](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130042/) failed e350d196d0b32add07b834e8c6378174a19ba4c9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130042/","number":"130042","commit":"e350d196d0b32add07b834e8c6378174a19ba4c9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130052/","number":"130052","commit":"4b192bbaa731f93683194bdb5fab5e6d8a8f4b74"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130087/","number":"130087","commit":"75b8da7982416bedef0078116a0e6a04c1b65d8d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130093/","number":"130093","commit":"d1f9231909e8e191114dce565cabe165373cd75b"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130197/)
* Commit: df38a261e0be61d22989dd20f3546e3bc390ec5b
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130197/execution/node/409/log)

### History
* :broken_heart: [Build #130093](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130093/) failed d1f9231909e8e191114dce565cabe165373cd75b
* :broken_heart: [Build #130087](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130087/) failed 75b8da7982416bedef0078116a0e6a04c1b65d8d
* :broken_heart: [Build #130052](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130052/) failed 4b192bbaa731f93683194bdb5fab5e6d8a8f4b74
* :broken_heart: [Build #130042](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130042/) failed e350d196d0b32add07b834e8c6378174a19ba4c9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130042/","number":"130042","commit":"e350d196d0b32add07b834e8c6378174a19ba4c9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130052/","number":"130052","commit":"4b192bbaa731f93683194bdb5fab5e6d8a8f4b74"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130087/","number":"130087","commit":"75b8da7982416bedef0078116a0e6a04c1b65d8d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130093/","number":"130093","commit":"d1f9231909e8e191114dce565cabe165373cd75b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130197/","number":"130197","commit":"df38a261e0be61d22989dd20f3546e3bc390ec5b"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130443/)
* Commit: 54f02f3516a4a00a0aa023b37898faaec640294f
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101718/54f02f3516a4a00a0aa023b37898faaec640294f)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/52d62ceec55ce645113c9ea2d711eb2118982437) | [after](https://github.com/elastic/kibana/commit/54f02f3516a4a00a0aa023b37898faaec640294f) | diff |
| --- | --- | --- | --- |
| `lens` | 670 | 685 | +15 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/52d62ceec55ce645113c9ea2d711eb2118982437) | [after](https://github.com/elastic/kibana/commit/54f02f3516a4a00a0aa023b37898faaec640294f) | diff |
| --- | --- | --- | --- |
| `canvas` | 1.3MB | 1.3MB | +315.0B |
| `kibanaReact` | 312.0KB | 312.1KB | +45.0B |
| `lens` | 1.3MB | 1.4MB | +89.9KB |
| total |  |  | +90.2KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/52d62ceec55ce645113c9ea2d711eb2118982437) | [after](https://github.com/elastic/kibana/commit/54f02f3516a4a00a0aa023b37898faaec640294f) | diff |
| --- | --- | --- | --- |
| `expressions` | 199.2KB | 199.6KB | +332.0B |
| `kbnUiSharedDeps-css` | 503.3KB | 503.7KB | +409.0B |
| `kbnUiSharedDeps-elastic` | 2.6MB | 2.6MB | +1.0B |
| `kbnUiSharedDeps-js` | 6.4MB | 6.4MB | +5.4KB |
| `lens` | 38.1KB | 38.1KB | +69.0B |
| total |  |  | +6.2KB |

### History
* :broken_heart: [Build #130213](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130213/) failed 6997ed306b0282b046af599cf9157cd7123010cb
* :broken_heart: [Build #130197](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130197/) failed df38a261e0be61d22989dd20f3546e3bc390ec5b
* :broken_heart: [Build #130093](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130093/) failed d1f9231909e8e191114dce565cabe165373cd75b
* :broken_heart: [Build #130087](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130087/) failed 75b8da7982416bedef0078116a0e6a04c1b65d8d
* :broken_heart: [Build #130052](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130052/) failed 4b192bbaa731f93683194bdb5fab5e6d8a8f4b74

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130052/","number":"130052","commit":"4b192bbaa731f93683194bdb5fab5e6d8a8f4b74"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130087/","number":"130087","commit":"75b8da7982416bedef0078116a0e6a04c1b65d8d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130093/","number":"130093","commit":"d1f9231909e8e191114dce565cabe165373cd75b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130197/","number":"130197","commit":"df38a261e0be61d22989dd20f3546e3bc390ec5b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130213/","number":"130213","commit":"6997ed306b0282b046af599cf9157cd7123010cb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130443/","number":"130443","commit":"54f02f3516a4a00a0aa023b37898faaec640294f"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130498/)
* Commit: af45859d90400ad56f8539606fb02dc739eb43fe
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101718/af45859d90400ad56f8539606fb02dc739eb43fe)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/fe3c761b4056819abab5a943071709f9cd10871a) | [after](https://github.com/elastic/kibana/commit/af45859d90400ad56f8539606fb02dc739eb43fe) | diff |
| --- | --- | --- | --- |
| `lens` | 687 | 688 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/fe3c761b4056819abab5a943071709f9cd10871a) | [after](https://github.com/elastic/kibana/commit/af45859d90400ad56f8539606fb02dc739eb43fe) | diff |
| --- | --- | --- | --- |
| `lens` | 1.4MB | 1.4MB | +4.6KB |

### History
* :green_heart: [Build #130443](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130443/) succeeded 54f02f3516a4a00a0aa023b37898faaec640294f
* :broken_heart: [Build #130213](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130213/) failed 6997ed306b0282b046af599cf9157cd7123010cb
* :broken_heart: [Build #130197](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130197/) failed df38a261e0be61d22989dd20f3546e3bc390ec5b
* :broken_heart: [Build #130093](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130093/) failed d1f9231909e8e191114dce565cabe165373cd75b
* :broken_heart: [Build #130087](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130087/) failed 75b8da7982416bedef0078116a0e6a04c1b65d8d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130087/","number":"130087","commit":"75b8da7982416bedef0078116a0e6a04c1b65d8d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130093/","number":"130093","commit":"d1f9231909e8e191114dce565cabe165373cd75b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130197/","number":"130197","commit":"df38a261e0be61d22989dd20f3546e3bc390ec5b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130213/","number":"130213","commit":"6997ed306b0282b046af599cf9157cd7123010cb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130443/","number":"130443","commit":"54f02f3516a4a00a0aa023b37898faaec640294f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130498/","number":"130498","commit":"af45859d90400ad56f8539606fb02dc739eb43fe"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-app (Team:KibanaApp)

flash1293 commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130859/)
* Commit: a5d9b924d797cd9836874b210b5501090b64786c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101718/a5d9b924d797cd9836874b210b5501090b64786c)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/80b109f95c344c8e7f8d484242f4be2feb32ba3c) | [after](https://github.com/elastic/kibana/commit/a5d9b924d797cd9836874b210b5501090b64786c) | diff |
| --- | --- | --- | --- |
| `lens` | 687 | 688 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/80b109f95c344c8e7f8d484242f4be2feb32ba3c) | [after](https://github.com/elastic/kibana/commit/a5d9b924d797cd9836874b210b5501090b64786c) | diff |
| --- | --- | --- | --- |
| `lens` | 1.4MB | 1.4MB | +4.6KB |

### History
* :green_heart: [Build #130498](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130498/) succeeded af45859d90400ad56f8539606fb02dc739eb43fe
* :green_heart: [Build #130443](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130443/) succeeded 54f02f3516a4a00a0aa023b37898faaec640294f
* :broken_heart: [Build #130213](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130213/) failed 6997ed306b0282b046af599cf9157cd7123010cb
* :broken_heart: [Build #130197](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130197/) failed df38a261e0be61d22989dd20f3546e3bc390ec5b
* :broken_heart: [Build #130093](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130093/) failed d1f9231909e8e191114dce565cabe165373cd75b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130093/","number":"130093","commit":"d1f9231909e8e191114dce565cabe165373cd75b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130197/","number":"130197","commit":"df38a261e0be61d22989dd20f3546e3bc390ec5b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130213/","number":"130213","commit":"6997ed306b0282b046af599cf9157cd7123010cb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130443/","number":"130443","commit":"54f02f3516a4a00a0aa023b37898faaec640294f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130498/","number":"130498","commit":"af45859d90400ad56f8539606fb02dc739eb43fe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130859/","number":"130859","commit":"a5d9b924d797cd9836874b210b5501090b64786c"}]}
PIPELINE-->

wylieconlon commented on x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/dimension_editor.tsx:
```diff
@@ -603,6 +607,8 @@ export function DimensionEditor(props: DimensionEditorProps) {
     <ParamEditor
       layer={state.layers[layerId]}
+      layerId={layerId}
+      activeData={props.activeData}
```
This looks like it'll cause extra re-renders of the formula panel and all the others, any idea how much of an impact this has?

wylieconlon commented on x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/formula/editor/formula_help.tsx:
```diff
@@ -391,6 +391,14 @@ export function getFunctionSignatureLabel(
+      extraArgs += i18n.translate('xpack.lens.formula.shiftExtraArguments', {
+        defaultMessage: '[shift]?: string',
+      });
+    }
```
The logic here breaks when transitioning to a shifted count when using the `en-xa` locale, it produces `count(Recordsshift='3h')` instead of `count(shift='3h')`. Maybe we're doing some string equality?

wylieconlon reviewed (commented):
I'm seeing an inconsistent warning count when transitioning from a Count with shift to a formula, then changing the interval:

<img width="491" alt="Screen Shot 2021-06-14 at 4 30 48 PM" src="https://user-images.githubusercontent.com/666475/121955546-16d24200-cd2e-11eb-8bbf-ac9009047826.png">

dej611 reviewed (commented):
<img width="448" alt="Screenshot 2021-06-15 at 11 42 11" src="https://user-images.githubusercontent.com/924948/122032266-c0b5db00-cdcf-11eb-9d1c-681204a4f3ea.png">

Because now an extra argument is allowed on functions, the `lucene` suggestion is coming even where a `kql` query has been already put in place.
Would it be possible to remove it? There's already a validation error, but it would be nice to not promote it.

Also, I was wondering if we could improve something on this side: not a blocker but the helping tooltip is not pointing to the right direction at least.
<img width="347" alt="Screenshot 2021-06-15 at 11 41 19" src="https://user-images.githubusercontent.com/924948/122032622-1b4f3700-cdd0-11eb-88ed-10223df35c8a.png">

flash1293 commented:
Addressed the following points:
* Did the same warnings for formula and chart
* Calculate the important bit (date histogram interval) outside of the formula editor boundary
* Fixed the logic for looking up the field name - it was looking for a hard-coded `"Records"` - changed that to look at the operation type instead
* Do not show `lucene` if `kql` got added already
* Improve error message for invalid time shift in the formula

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131405/)
* Commit: cc16b186e65b4bba89c3b921370375f18d7870ce
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101718/cc16b186e65b4bba89c3b921370375f18d7870ce)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup13](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131405/execution/node/783/log)
* [Execute xpack-kibana-ciGroup13](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131405/execution/node/831/log)

### History
* :green_heart: [Build #130859](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130859/) succeeded a5d9b924d797cd9836874b210b5501090b64786c
* :green_heart: [Build #130498](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130498/) succeeded af45859d90400ad56f8539606fb02dc739eb43fe
* :green_heart: [Build #130443](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130443/) succeeded 54f02f3516a4a00a0aa023b37898faaec640294f
* :broken_heart: [Build #130213](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130213/) failed 6997ed306b0282b046af599cf9157cd7123010cb
* :broken_heart: [Build #130197](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130197/) failed df38a261e0be61d22989dd20f3546e3bc390ec5b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130197/","number":"130197","commit":"df38a261e0be61d22989dd20f3546e3bc390ec5b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130213/","number":"130213","commit":"6997ed306b0282b046af599cf9157cd7123010cb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130443/","number":"130443","commit":"54f02f3516a4a00a0aa023b37898faaec640294f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130498/","number":"130498","commit":"af45859d90400ad56f8539606fb02dc739eb43fe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130859/","number":"130859","commit":"a5d9b924d797cd9836874b210b5501090b64786c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131405/","number":"131405","commit":"cc16b186e65b4bba89c3b921370375f18d7870ce"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131402/)
* Commit: 51e0afd925ef7a26852efae59131fd76c6fbcb8f
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101718/51e0afd925ef7a26852efae59131fd76c6fbcb8f)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup13](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131402/execution/node/788/log)
* [Execute xpack-kibana-ciGroup13](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131402/execution/node/831/log)

### History
* :broken_heart: [Build #131405](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131405/) failed cc16b186e65b4bba89c3b921370375f18d7870ce
* :green_heart: [Build #130859](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130859/) succeeded a5d9b924d797cd9836874b210b5501090b64786c
* :green_heart: [Build #130498](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130498/) succeeded af45859d90400ad56f8539606fb02dc739eb43fe
* :green_heart: [Build #130443](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130443/) succeeded 54f02f3516a4a00a0aa023b37898faaec640294f
* :broken_heart: [Build #130213](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130213/) failed 6997ed306b0282b046af599cf9157cd7123010cb

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130213/","number":"130213","commit":"6997ed306b0282b046af599cf9157cd7123010cb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130443/","number":"130443","commit":"54f02f3516a4a00a0aa023b37898faaec640294f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130498/","number":"130498","commit":"af45859d90400ad56f8539606fb02dc739eb43fe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130859/","number":"130859","commit":"a5d9b924d797cd9836874b210b5501090b64786c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131405/","number":"131405","commit":"cc16b186e65b4bba89c3b921370375f18d7870ce"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131402/","number":"131402","commit":"51e0afd925ef7a26852efae59131fd76c6fbcb8f"}]}
PIPELINE-->

flash1293 commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131434/)
* Commit: f08b38b021cffdcf13d403c62992dd233b9c4af6
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101718/f08b38b021cffdcf13d403c62992dd233b9c4af6)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/2c17897f062cdbb9ddbd1553f3c2c1b73eeba808) | [after](https://github.com/elastic/kibana/commit/f08b38b021cffdcf13d403c62992dd233b9c4af6) | diff |
| --- | --- | --- | --- |
| `lens` | 687 | 688 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/2c17897f062cdbb9ddbd1553f3c2c1b73eeba808) | [after](https://github.com/elastic/kibana/commit/f08b38b021cffdcf13d403c62992dd233b9c4af6) | diff |
| --- | --- | --- | --- |
| `lens` | 1.4MB | 1.4MB | +4.7KB |

### History
* :broken_heart: [Build #131402](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131402/) failed 51e0afd925ef7a26852efae59131fd76c6fbcb8f
* :broken_heart: [Build #131405](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131405/) failed cc16b186e65b4bba89c3b921370375f18d7870ce
* :green_heart: [Build #130859](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130859/) succeeded a5d9b924d797cd9836874b210b5501090b64786c
* :green_heart: [Build #130498](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130498/) succeeded af45859d90400ad56f8539606fb02dc739eb43fe
* :green_heart: [Build #130443](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130443/) succeeded 54f02f3516a4a00a0aa023b37898faaec640294f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130443/","number":"130443","commit":"54f02f3516a4a00a0aa023b37898faaec640294f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130498/","number":"130498","commit":"af45859d90400ad56f8539606fb02dc739eb43fe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130859/","number":"130859","commit":"a5d9b924d797cd9836874b210b5501090b64786c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131405/","number":"131405","commit":"cc16b186e65b4bba89c3b921370375f18d7870ce"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131402/","number":"131402","commit":"51e0afd925ef7a26852efae59131fd76c6fbcb8f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131434/","number":"131434","commit":"f08b38b021cffdcf13d403c62992dd233b9c4af6"}]}
PIPELINE-->

flash1293 commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131596/)
* Commit: 989b8991d51f4b00d477ee78b293a8a60bcd3501
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101718/989b8991d51f4b00d477ee78b293a8a60bcd3501)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/aea2a384fd49e187dfc5edcf365faa0726d0b7ce) | [after](https://github.com/elastic/kibana/commit/989b8991d51f4b00d477ee78b293a8a60bcd3501) | diff |
| --- | --- | --- | --- |
| `lens` | 690 | 691 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/aea2a384fd49e187dfc5edcf365faa0726d0b7ce) | [after](https://github.com/elastic/kibana/commit/989b8991d51f4b00d477ee78b293a8a60bcd3501) | diff |
| --- | --- | --- | --- |
| `lens` | 1.5MB | 1.5MB | +4.7KB |

### History
* :green_heart: [Build #131434](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131434/) succeeded f08b38b021cffdcf13d403c62992dd233b9c4af6
* :broken_heart: [Build #131402](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131402/) failed 51e0afd925ef7a26852efae59131fd76c6fbcb8f
* :broken_heart: [Build #131405](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131405/) failed cc16b186e65b4bba89c3b921370375f18d7870ce
* :green_heart: [Build #130859](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130859/) succeeded a5d9b924d797cd9836874b210b5501090b64786c
* :green_heart: [Build #130498](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130498/) succeeded af45859d90400ad56f8539606fb02dc739eb43fe

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130498/","number":"130498","commit":"af45859d90400ad56f8539606fb02dc739eb43fe"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130859/","number":"130859","commit":"a5d9b924d797cd9836874b210b5501090b64786c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131405/","number":"131405","commit":"cc16b186e65b4bba89c3b921370375f18d7870ce"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131402/","number":"131402","commit":"51e0afd925ef7a26852efae59131fd76c6fbcb8f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131434/","number":"131434","commit":"f08b38b021cffdcf13d403c62992dd233b9c4af6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131596/","number":"131596","commit":"989b8991d51f4b00d477ee78b293a8a60bcd3501"}]}
PIPELINE-->

wylieconlon reviewed (approved):
LGTM, the performance impact of the extra re-renders appears to be minimal.

flash1293 commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131953/)
* Commit: 479cd300cff88a7daa477888409ef308d33c83b3
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101718/479cd300cff88a7daa477888409ef308d33c83b3)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/fd5883b4654ce574837f79c529d38b5f976232e8) | [after](https://github.com/elastic/kibana/commit/479cd300cff88a7daa477888409ef308d33c83b3) | diff |
| --- | --- | --- | --- |
| `lens` | 690 | 691 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/fd5883b4654ce574837f79c529d38b5f976232e8) | [after](https://github.com/elastic/kibana/commit/479cd300cff88a7daa477888409ef308d33c83b3) | diff |
| --- | --- | --- | --- |
| `lens` | 1.5MB | 1.5MB | +4.7KB |

### History
* :green_heart: [Build #131596](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131596/) succeeded 989b8991d51f4b00d477ee78b293a8a60bcd3501
* :green_heart: [Build #131434](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131434/) succeeded f08b38b021cffdcf13d403c62992dd233b9c4af6
* :broken_heart: [Build #131402](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131402/) failed 51e0afd925ef7a26852efae59131fd76c6fbcb8f
* :broken_heart: [Build #131405](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131405/) failed cc16b186e65b4bba89c3b921370375f18d7870ce
* :green_heart: [Build #130859](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130859/) succeeded a5d9b924d797cd9836874b210b5501090b64786c

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130859/","number":"130859","commit":"a5d9b924d797cd9836874b210b5501090b64786c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131405/","number":"131405","commit":"cc16b186e65b4bba89c3b921370375f18d7870ce"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131402/","number":"131402","commit":"51e0afd925ef7a26852efae59131fd76c6fbcb8f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131434/","number":"131434","commit":"f08b38b021cffdcf13d403c62992dd233b9c4af6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131596/","number":"131596","commit":"989b8991d51f4b00d477ee78b293a8a60bcd3501"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131953/","number":"131953","commit":"479cd300cff88a7daa477888409ef308d33c83b3"}]}
PIPELINE-->

dej611 reviewed (approved):
Code LGTM. Tested again and found no more issues 👍

kibanamachine commented:
## 💚 Backport successful

| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [7.x](https://github.com/elastic/kibana/pull/102480)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/102480">](https://github.com/elastic/kibana/pull/102480) |

This backport PR will be merged automatically after passing CI.
